lucet-module/src/bindings.rs

```diff
@@ -76,13 +75,15 @@ impl Bindings {
         match self.bindings.get(module) {
             Some(m) => match m.get(symbol) {
                 Some(s) => Ok(s),
-                None => Err(format_err!("Unknown symbol `{}::{}`", module, symbol)),
+                None => Err(Error::UnknownSymbol {
+                    module: module.to_string(),
```

Comment:
@acfoltzer -- I took an initial pass at converting all of the errors in bindings.rs to fields of the enum in errors.rs.  I'm unhappy with my use of to_string() and I wonder if you might have thoughts on a better way.

Reply:
Adam might have other opinions but `.to_string` looks okay to me :slightly_smiling_face:

Reply:
`format!("{}", x) ≃ x.to_string()`, so this looks fine to me!

Reply:
IMO `module.to_owned()` is a better fit here, as explained in:
https://users.rust-lang.org/t/to-string-vs-to-owned-for-string-literals/1441/6

Reply:
Thanks!